src/transport_picoquic.cpp

```diff
@@ -706,9 +712,20 @@ PicoQuicTransport::Close(const TransportConnId& conn_id, uint64_t app_reason_cod
     }
 
     // Only one datagram context is per connection, if it's deleted, then the connection is to be terminated
-    OnConnectionStatus(conn_id, TransportStatus::kDisconnected);
+    switch (app_reason_code) {
+        case 1: // idle timeout
+            OnConnectionStatus(conn_id, TransportStatus::kIdleTimeout);
+            break;
+
+        case 100: // Client shutting down connection
```

Comment:
Should we formalize these error codes into an enum class in another PR?

Reply:
Yeah, we need app reason codes that we support in addition to the ones that MoQT defines. I created https://github.com/Quicr/libquicr/issues/247 to track and fix this.